File: io-hotmoka-network/src/main/java/io/hotmoka/network/models/requests/NonInitialTransactionRequestModel.java
package io.hotmoka.network.models.requests;

import io.hotmoka.beans.requests.NonInitialTransactionRequest;
import io.hotmoka.network.models.values.StorageReferenceModel;
import io.hotmoka.network.models.values.TransactionReferenceModel;

public abstract class NonInitialTransactionRequestModel extends TransactionRequestModel {
    public StorageReferenceModel caller;
    public String nonce;
    public TransactionReferenceModel classpath;
    public String gasLimit;
    public String gasPrice;

    protected NonInitialTransactionRequestModel(NonInitialTransactionRequest<?> request) {
    	this.caller = new StorageReferenceModel(request.caller);
    	this.nonce = request.nonce.toString();
    	this.classpath = new TransactionReferenceModel(request.classpath);
    	this.gasLimit = request.gasLimit.toString();
    	this.gasPrice = request.gasPrice.toString();
    }

    protected NonInitialTransactionRequestModel() {}
}

File: io-hotmoka-network-thin-client/io-hotmoka-network-thin-client-kt/src/main/kotlin/io/hotmoka/network/thin/client/models/requests/NonInitialTransactionRequestModel.kt
package io.hotmoka.network.thin.client.models.requests

import io.hotmoka.network.thin.client.models.values.StorageReferenceModel
import io.hotmoka.network.thin.client.models.values.TransactionReferenceModel

abstract class NonInitialTransactionRequestModel(
    val signature: String,
    val caller: StorageReferenceModel,
    val nonce: String,
    val classpath: TransactionReferenceModel,
    val chainId: String,
    val gasLimit: String,
    val gasPrice: String
): TransactionRequestModel()